docs/SupportedONNXOps-NNPA.md

```diff
@@ -1,5 +1,5 @@
 <!--- Automatically generated, do not edit. -->
-<!--- To update, run `make onnx_mlir_supported_ops` -->
+<!--- python3 documentOps.py --arch NNPA --input ../test/accelerators/NNPA/backend/CMakeLists.txt --path . --notes -->
```

Comment:
If `make onnx_mlir_supported_ops_NNPA` works, I think the place to edit is in `documentOps.py`: https://github.com/onnx/onnx-mlir/blob/main/utils/documentOps.py#L208, that generated this text `To update, run make onnx_mlir_supported_ops`

Reply:
Okay let me try that then. I tried `onnx_mlir_supported_ops` and that did not work... I was replacing the command with what was previously there but I can try it.

Reply:
@tungld Unfortunately this is what happens when I run the command `make onnx_mlir_supported_ops`:
Feel free to try for yourself if need be :) 

```
<!--- Automatically generated, do not edit. -->
<!--- To update, run `make onnx_mlir_supported_ops` -->

# Supported ONNX Operation for Target *NNPA*.

Onnx-mlir currently supports ONNX operations targeting up to opset 19. Limitations are listed when applicable. This documentation highlights the minimum and maximum opset versions that are fully supported by onnx-mlir and not the version changes.

* Operations are defined by the [ONNX Standard](https://github.com/onnx/onnx/blob/main/docs/Operators.md).
* **Supported Opsets** indicates the lowest and highest opset a model may have for onnx-mlir to support compiling a model with the operator. 
   * A * indicates onnx-mlir is compatible with the latest version of that operator available as of opset 19.


NNPA has hardware limitations in dimension index size and tensor size, which are described in [NNPALimit.h](../src/Accelerators/NNPA/Conversion/ONNXToZHigh/NNPALimit.h). They are large enough for normal use cases, but if your model exceeds the limitations, CPU is used instead of NNPA.


| Op |Supported Opsets (inclusive) |Limitations |Notes |
| --- |--- |--- |--- |
| **Abs** |none | | | |
| **Acos** |none | | | |
| **Acosh** |none | | | |
| **Adagrad** |none | | | |
| **Adam** |none | | | |
| **Add** |6 - * |- Shape of input tensors must be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
| **And** |none | | | |
| **ArgMax** |none | | | |
| **ArgMin** |none | | | |
| **ArrayFeatureExtractor** |none | | | |
| **Asin** |none | | | |
| **Asinh** |none | | | |
| **Atan** |none | | | |
| **Atanh** |none | | | |
| **AveragePool** |6 - * |- `auto_pad` must be `NOTSET`, `VALID`, and `SAME_UPPER`. If `NOTSET` is used, `pads` must be set so that the padding valid type or same upper.<br>- `ceil_mode` must be default value(0) <br>- Input and output tensors must be 4D tensors (N x C x H x W).<br>- `kernel_shape` must be static.<br>- `count_include_pad` must be default value(0).<br>- `ceil_mode` must be default value(0). | |
| **BatchNormalization** |6 - * |Input and output tensor must be 4D(N x C x H x W). | |
| **Bernoulli** |none | | | |
| **Binarizer** |none | | | |
| **BitShift** |none | | | |
| **BitwiseAnd** |none | | | |
| **BitwiseNot** |none | | | |
| **BitwiseOr** |none | | | |
| **BitwiseXor** |none | | | |
| **BlackmanWindow** |none | | | |
| **Cast** |none | | | |
| **CastLike** |none | | | |
| **CastMap** |none | | | |
| **CategoryMapper** |none | | | |
| **Ceil** |none | | | |
| **Celu** |none | | | |
| **CenterCropPad** |none | | | |
| **Clip** |none | | | |
| **Col2Im** |none | | | |
| **Compress** |none | | | |
| **Concat** |none | | | |
| **ConcatFromSequence** |none | | | |
| **Constant** |none | | | |
| **ConstantOfShape** |none | | | |
| **Conv** |6 - * |- `auto_pad` must be `NOTSET`, `VALID`, and `SAME_UPPER`. If `NOTSET` is used, `pads` must be set so that the padding valid type or same upper.<br>- Dimension in Height and weight must be static.<br>- `group` must be default value(1).<br>- `dilations` must be default value(1).<br>- Input and output tensors must have 4D (N x C x H x W).<br>- `kernel_shape` must be static. | |
| **ConvInteger** |none | | | |
| **ConvTranspose** |6 - * |- 1D and 3D not supported because Conv1D and Conv3D not supported in zDNN. non-default `dilations` not supported because dilated convolution not supported in zDNN. | |
| **Cos** |none | | | |
| **Cosh** |none | | | |
| **CumSum** |none | | | |
| **DFT** |none | | | |
| **DeformConv** |none | | | |
| **DepthToSpace** |none | | | |
| **DequantizeLinear** |none | | | |
| **Det** |none | | | |
| **DictVectorizer** |none | | | |
| **Div** |6 - * |- Shape of input tensors must be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
| **Dropout** |none | | | |
| **DynamicQuantizeLinear** |none | | | |
| **Einsum** |none | | | |
| **Elu** |none | | | |
| **Equal** |none | | | |
| **Erf** |none | | | |
| **Exp** |6 - * |Input tensor must have 4 dimensions. | |
| **Expand** |none | | | |
| **EyeLike** |none | | | |
| **FeatureVectorizer** |none | | | |
| **Flatten** |none | | | |
| **Floor** |none | | | |
| **GRU** |7 - * |- `direction` and `hidden_size` in `W` must have static dimensions.<br>- `R` must have static dimensions.<br>- If `B` and `initial_h` are given, they must have static dimensions.<br>- `sequence_lens` is not supported.<br>- `activations` must be `["Sigmoid", "Tanh", "Tanh"]`.<br>- `clip` is not supported.<br>- `linear_before_reset` must be 1.<br>- `layout` is not supported. | |
| **Gather** |none | | | |
| **GatherElements** |none | | | |
| **GatherND** |none | | | |
| **Gemm** |6 - * |- `alpha` and `beta` must be default value(1).<br>- Rank of `C` must be 1 or 2. If the rank is 1, the dimension of `C` must be the same with the seconde dimension of `B`. | |
| **GlobalAveragePool** |6 - * |- Input shape must be 4D tensor(NCHW).<br>- Dimensions in `H` and `W` must be static. | |
| **GlobalLpPool** |none | | | |
| **GlobalMaxPool** |none | | | |
| **Gradient** |none | | | |
| **Greater** |none | | | |
| **GreaterOrEqual** |none | | | |
| **GridSample** |none | | | |
| **GroupNormalization** |none | | | |
| **HammingWindow** |none | | | |
| **HannWindow** |none | | | |
| **HardSigmoid** |none | | | |
| **HardSwish** |none | | | |
| **Hardmax** |none | | | |
| **Identity** |none | | | |
| **If** |none | | | |
| **Imputer** |none | | | |
| **InstanceNormalization** |none | | | |
| **IsInf** |none | | | |
| **IsNaN** |none | | | |
| **LRN** |none | | | |
| **LSTM** |7 - * |- `direction` and `hidden_size` in `W` must have static dimensions.<br>- `R` must have static dimensions.<br>- `B` and `initial_h` have static dimensions if given. `B`'s direction dim must be 1 or 2.<br>- `P`(peepholes), `activation_alpha`, and `activation_beta` are not supported.<br>- `activations` must be `["Sigmoid", "Tanh", "Tanh"]`.<br>- `clip` is not supported.<br>- `input_forget` must be default value(0).<br>- `layout` is not supported. | |
| **LabelEncoder** |none | | | |
| **LayerNormalization** |none | | | |
| **LeakyRelu** |6 - * |The operations immediately before and after the LeakyRelu operation must be executed on the NNPA. Otherwise, LeakyRelu is executed on the CPU. This limitation is set to avoid performance degradation. | |
| **Less** |none | | | |
| **LessOrEqual** |none | | | |
| **LinearClassifier** |none | | | |
| **LinearRegressor** |none | | | |
| **Log** |6 - * |Input tensor must have 4 dimensions. | |
| **LogSoftmax** |6 - * | | |
| **Loop** |none | | | |
| **LpNormalization** |none | | | |
| **LpPool** |none | | | |
| **MatMul** |6 - * |Ranks of input tensors must be (Rank of A, Rank of B) = (M, N), where M >= 2 and N >= 2. | |
| **MatMulInteger** |none | | | |
| **Max** |6 - * |- Shape of input tensors must be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
| **MaxPool** |6 - * |- `auto_pad` must be `NOTSET`, `VALID`, and `SAME_UPPER`. If `NOTSET` is used, `pads` must be set so that the padding valid type or same upper.<br>- `ceil_mode` must be default value(0) <br>- Input and output tensors must be 4D tensors(N x C x H x W).<br>- `kernel_shape` must be static.<br>- `ceil_mode` must be default value(0).<br>- `dilations` must be default value(1). | |
| **MaxRoiPool** |none | | | |
| **MaxUnpool** |none | | | |
| **Mean** |none | | | |
| **MeanVarianceNormalization** |none | | | |
| **MelWeightMatrix** |none | | | |
| **Min** |6 - * |- Shape of input tensors must be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
| **Mish** |none | | | |
| **Mod** |none | | | |
| **Momentum** |none | | | |
| **Mul** |6 - * |- Shape of input tensors should be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
| **Multinomial** |none | | | |
| **Neg** |none | | | |
| **NegativeLogLikelihoodLoss** |none | | | |
| **NonMaxSuppression** |none | | | |
| **NonZero** |none | | | |
| **Normalizer** |none | | | |
| **Not** |none | | | |
| **OneHot** |none | | | |
| **OneHotEncoder** |none | | | |
| **Optional** |none | | | |
| **OptionalGetElement** |none | | | |
| **OptionalHasElement** |none | | | |
| **Or** |none | | | |
| **PRelu** |none | | | |
| **Pad** |none | | | |
| **Pow** |7 - * |- Exponent should be a scalar integer and less or equal to 64. | |
| **QLinearConv** |none | | | |
| **QLinearMatMul** |none | | | |
| **QuantizeLinear** |none | | | |
| **RNN** |none | | | |
| **RandomNormal** |none | | | |
| **RandomNormalLike** |none | | | |
| **RandomUniform** |none | | | |
| **RandomUniformLike** |none | | | |
| **Range** |none | | | |
| **Reciprocal** |none | | | |
| **ReduceL1** |none | | | |
| **ReduceL2** |none | | | |
| **ReduceLogSum** |none | | | |
| **ReduceLogSumExp** |none | | | |
| **ReduceMax** |none | | | |
| **ReduceMean** |6 - * |- `keepdims` must be 1.<br>- Input tensor must be 4D tensors and `axis` must be [2, 3]. | |
| **ReduceMin** |none | | | |
| **ReduceProd** |none | | | |
| **ReduceSum** |none | | | |
| **ReduceSumSquare** |none | | | |
| **Relu** |6 - * |Input tensor must be less than or equal to 4 dimensions. | |
| **Reshape** |none | | | |
| **Resize** |none | | | |
| **ReverseSequence** |none | | | |
| **RoiAlign** |none | | | |
| **Round** |none | | | |
| **STFT** |none | | | |
| **SVMClassifier** |none | | | |
| **SVMRegressor** |none | | | |
| **Scaler** |none | | | |
| **Scan** |none | | | |
| **Scatter** |none | | | |
| **ScatterElements** |none | | | |
| **ScatterND** |none | | | |
| **Selu** |none | | | |
| **SequenceAt** |none | | | |
| **SequenceConstruct** |none | | | |
| **SequenceEmpty** |none | | | |
| **SequenceErase** |none | | | |
| **SequenceInsert** |none | | | |
| **SequenceLength** |none | | | |
| **SequenceMap** |none | | | |
| **Shape** |none | | | |
| **Shrink** |none | | | |
| **Sigmoid** |6 - * |Input tensor must be less than or equal to 4 dimensions. | |
| **Sign** |none | | | |
| **Sin** |none | | | |
| **Sinh** |none | | | |
| **Size** |none | | | |
| **Slice** |none | | | |
| **Softmax** |6 - * |- `axis` must be the last dimension, i.e. `rank - 1` or -1. | |
| **SoftmaxCrossEntropyLoss** |none | | | |
| **Softplus** |none | | | |
| **Softsign** |none | | | |
| **SpaceToDepth** |none | | | |
| **Split** |none | | | |
| **SplitToSequence** |none | | | |
| **Sqrt** |none | | | |
| **Squeeze** |none | | | |
| **StringNormalizer** |none | | | |
| **Sub** |6 - * |- Shape of input tensors should be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
| **Sum** |6 - * |- must have the same static shape (Broadcasting not supported.)<br>- Single input not supported. | |
| **Tan** |none | | | |
| **Tanh** |6 - * |Input tensor must be less than or equal to 4 dimensions. | |
| **TfIdfVectorizer** |none | | | |
| **ThresholdedRelu** |none | | | |
| **Tile** |none | | | |
| **TopK** |none | | | |
| **Transpose** |none | | | |
| **TreeEnsembleClassifier** |none | | | |
| **TreeEnsembleRegressor** |none | | | |
| **Trilu** |none | | | |
| **Unique** |none | | | |
| **Unsqueeze** |none | | | |
| **Upsample** |none | | | |
| **Where** |none | | | |
| **Xor** |none | | | |
| **ZipMap** |none | | | |

```